[#8] Color, Typography 공통 컴포넌트를 새 디자인에 맞게 수정

cskime opened the pull request:
## 📝 작업 내용

- `Color`, `Typography` 공통 컴포넌트를 새 디자인에 맞게 수정합니다.
- 기존에 구현했던 방식에서 사용성을 좀 더 개선했습니다.
- `red` 색상은 삭제 버튼에 사용되는데, 버튼의 hover, active 등 상태에 따라 변경되는 색상에 사용하기 위해 임의로 shade 100~900 색상을 만들었습니다.


## 📷 스크린샷 (선택)

<img width="1024" height="953" alt="image" src="https://github.com/user-attachments/assets/fe510815-c173-4933-a90a-0b7d1b17b0e2" />

cskime committed:
♻️ refactor [#8] Typography를 새 디자인에 맞게 수정

cskime committed:
✨ feat [#8] Typography class name을 JS 객체로 선언

- CSS 파일을 직접 import하면 import 변수 이름을 임의로 변경할 수 있습니다.
- 항상 `Typography` 라는 이름을 통해 class name을 사용할 수 있도록 JS 객체로 wrapping 합니다.

cskime committed:
✨ feat [#8] Component test page에서 Typography 예시 코드 변경

cskime committed:
✨ feat [#8] 새 디자인에 정의된 색상으로 변경

cskime committed:
✨ type [#8] Component test page에서 Color 예시 코드 변경

YOOJEYEONG commented on src/pages/test-components/index.tsx:
테스트 페이지 부분은 저도 버튼 컴포넌트를 올려놔서 충돌날거 같습니다.

cskime replied:
넵 병합할 때 신경 써야겠네요

cskime committed:
✨ feat [#8] Red 색상에 100~900 shade를 적용한 색상 추가

- `red` 색상은 삭제 버튼에 사용됩니다.
- 버튼의 hover, active 등 상태에 따라 변경되는 색상에 사용하기 위해 임의로 shade 100~900 색상을 만들었습니다.

cskime committed:
Merge remote-tracking branch 'origin/develop' into feature/#8

Merge conflict 수정
